src/Tools/dotnet-dump/Analyzer.cs

```diff
@@ -25,45 +26,60 @@ public async Task<int> Analyze(FileInfo dump_path, string[] command)
         {
             _consoleProvider.Out.WriteLine($"Loading core dump: {dump_path} ...");
 
-            DataTarget target = null;
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux)) {
-                target = DataTarget.LoadCoreDump(dump_path.FullName);
-            }
-            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) {
-                target = DataTarget.LoadCrashDump(dump_path.FullName, CrashDumpReader.ClrMD);
-            }
-            else {
-                _consoleProvider.Error.WriteLine($"{RuntimeInformation.OSDescription} not supported");
-                return 1;
-            }
-
-            using (target)
-            {
-                // Create common analyze context for commands
-                var analyzeContext = new AnalyzeContext(_consoleProvider, target, _consoleProvider.Stop) {
-                    CurrentThreadId = unchecked((int)target.DataReader.EnumerateAllThreads().FirstOrDefault())
-                };
-                _commandProcessor.CommandContext = analyzeContext;
-
-                // Automatically enable symbol server support on Linux and MacOS
-                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) {
-                    await _commandProcessor.Parse("setsymbolserver -ms", _consoleProvider);
+            try
+            { 
+                DataTarget target = null;
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux)) {
+                    target = DataTarget.LoadCoreDump(dump_path.FullName);
+                }
+                else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) {
+                    target = DataTarget.LoadCrashDump(dump_path.FullName, CrashDumpReader.ClrMD);
+                }
+                else {
+                    throw new PlatformNotSupportedException($"Unsupported operating system: {RuntimeInformation.OSDescription}");
                 }
 
-                // Run the commands from the dotnet-dump command line
-                if (command != null)
+                using (target)
                 {
-                    foreach (string cmd in command)
+                    // Create common analyze context for commands
+                    var analyzeContext = new AnalyzeContext(_consoleProvider, target, _consoleProvider.Stop) {
+                        CurrentThreadId = unchecked((int)target.DataReader.EnumerateAllThreads().FirstOrDefault())
+                    };
+                    _commandProcessor.CommandContext = analyzeContext;
+
+                    // Automatically enable symbol server support on Linux and MacOS
+                    if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) {
+                        await _commandProcessor.Parse("setsymbolserver -ms", _consoleProvider);
```

Comment:
Nit: Is there a reason we can't directly call the API rather than parsing a string?

Reply:
Yes it can.